dep0/lib/01_ast/include/dep0/ast/rename.hpp

```diff
@@ -5,11 +5,12 @@
 namespace dep0::ast {
 
 template <Properties P>
-typename type_t<P>::var_t rename(
-    typename type_t<P>::var_t const&,
-    typename type_t<P>::arr_t::arg_iterator begin,
-    typename type_t<P>::arr_t::arg_iterator end,
-    type_t<P>& ret_type);
+typename expr_t<P>::var_t rename(
+    typename expr_t<P>::var_t const&,
```

Comment:
All of these functions to rename , replace, substitute, etc should take var_t by value